src/Ethereum/Barz.cpp

```diff
@@ -72,9 +73,16 @@ Data getInitCodeFromAttestationObject(const std::string& factoryAddress, const s
     return getInitCodeFromPublicKey(factoryAddress, hexEncoded(publicKey), verificationFacet);
 }
 
-Data getFormattedSignature(const Data& signature, const Data& authenticatorData, const std::string& origin) {
-    const std::string clientDataJSONPre = "{\"type\":\"webauthn.get\",\"challenge\":\"";
-    const std::string clientDataJSONPost = "\",\"origin\":\"" + origin + "\"}";
+Data getFormattedSignature(const Data& signature, const Data challenge, const Data& authenticatorData, const std::string& clientDataJSON) {
+    std::string challengeBase64 = TW::Base64::encodeBase64Url(challenge);
+    challengeBase64.pop_back();
```

Comment:
Is there a better way to remove the last "=" from Base64 encoded result?

Reply:
Yes, "=" is always omitted in webauthn: https://w3c.github.io/webauthn/#base64url-encoding
Thanks, I'll update the code!